metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java

```diff
@@ -546,4 +562,14 @@ public PartitionDto get(final QualifiedName name) {
     public boolean exists(final QualifiedName name) {
         return get(name) != null;
     }
+
+    private TableInfo getTableInfo(final QualifiedName name) {
```

Comment:
We can as well pass TableDto instead of TableInfo? Also you can use the Optional method to return.

return tableService.get(name, GetTableServiceParameters.builder().includeInfo(true).useCache(true).build()).orElseThrow(..)

Reply:
Don't want to leak the TableDto into connector implementation. Will use Optional method.